src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/TempPE/TempPEBuildManager.cs

```diff
@@ -0,0 +1,315 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Threading.Tasks.Dataflow;
+using Microsoft.VisualStudio.ProjectSystem.LanguageServices;
+using Microsoft.VisualStudio.ProjectSystem.VS.Automation;
+using Microsoft.VisualStudio.Threading.Tasks;
+using VSLangProj;
+using InputTuple = System.Tuple<Microsoft.VisualStudio.ProjectSystem.IProjectSnapshot, Microsoft.VisualStudio.ProjectSystem.IProjectSubscriptionUpdate>;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.TempPE
+{
+    [Export(typeof(ITempPEBuildManager))]
+    internal class TempPEBuildManager : UnconfiguredProjectHostBridge<IProjectVersionedValue<InputTuple>, DesignTimeInputsDelta, IProjectVersionedValue<DesignTimeInputsItem>>, ITempPEBuildManager, IDisposable
+    {
+        private readonly IUnconfiguredProjectCommonServices _unconfiguredProjectServices;
+        private readonly ILanguageServiceHost _languageServiceHost;
+        //private readonly ITempPECompiler _compiler;
+
+        [ImportingConstructor]
+        public TempPEBuildManager(IProjectThreadingService threadingService,
+            IUnconfiguredProjectCommonServices unconfiguredProjectServices,
+            ILanguageServiceHost languageServiceHost
+            //,ITempPECompilerHost compilerHost
+            )
+             : base(threadingService.JoinableTaskContext)
+        {
+            _unconfiguredProjectServices = unconfiguredProjectServices;
+            _languageServiceHost = languageServiceHost;
+            //_compiler = compiler;
+            BuildManager = new OrderPrecedenceImportCollection<BuildManager>(projectCapabilityCheckProvider: _unconfiguredProjectServices.Project);
+        }
+
+        [ImportMany(typeof(BuildManager))]
+        internal OrderPrecedenceImportCollection<BuildManager> BuildManager { get; set; }
+
+        protected override Task DisposeCoreAsync(bool initialized)
+        {
+            if (AppliedValue != null)
+            {
+                // dispose any of our cancellation series we still have
+                foreach (var item in AppliedValue.Value.Inputs.Values)
```

Comment:
While I love `var` personally.  Its use in this repository is restricted to only cases where they type is apprent (such as `var f = new Foo();`)

Reply:
> While I love `var` personally.  Its use in this repository is restricted to only cases where they type is apprent

I don't know how I got it in my head but I thought we were var everywhere. Which is weird because personally I'm not a fan of var :P

Will definitely be changing this one!

Reply:
It is. Sometimes I open solutions that are not so pure as ours, and I had Message off as a result